### PR TITLE
Hopeful updates to Minish Cap cheevos.

### PR DESCRIPTION
**Pull request by wolfman2000**
A few things were noticed during my testing offline, so let's put them in the RA Script now. No changes have been published to RA yet.

* Converted all `ScreenId` calls to 16-bit BE format. I've taken enough notes to confirm that the "higher" byte is always the world, and the "lower" is the sub map. This should help maintain map consistency.
* Extracted the line of taking damage to a `TookDamage` function.
* Simplified the Bested Vaati cheevo. We only care about starting the event and ending the event.
* Add Music Protection on Bested Vaati so the cheevo trigger doesn't show on file select.
* Simplified the logic for Bested Big Green Chuchu. Also ensured the trigger icon would show.
* Simplified the logic for Bested Gleerok.
* Simplified the logic for Bested Mazaal.

There's more that can probably be done, but I'm presently on Hot Spot internet, so I can't spend too much time getting things updated to more modern standards.

### CODE REVIEW

File: RA Scripts/Legend of Zelda, The The Minish Cap.rascript [hunk 1]
```diff
@@ -600,22 +600,23 @@ function MusicId() => byte(0x029EF4)
 
 function CreditsLoaded() => MusicId() == 0x09 && Delta(MusicId()) == 0x00
 
-function ScreenIdByte() => byte(0x000BF4)
-function ScreenIdWord() => word(0x000BF4)
+function ScreenId() => word_be(0x000BF4)
```
Comment (MagmarFire) [inline RA Scripts/Legend of Zelda, The The Minish Cap.rascript:603]: For memory accessor wrappers, we want to use camelCase. For my scripts, I reserve PascalCase for computables and other functions.

File: RA Scripts/Legend of Zelda, The The Minish Cap.rascript [hunk 2]
```diff
@@ -600,22 +600,23 @@ function MusicId() => byte(0x029EF4)
...
+
+function TookDamage() => byte(0x00AAEA) < Delta(byte(0x00AAEA))
```
Comment (MagmarFire) [inline RA Scripts/Legend of Zelda, The The Minish Cap.rascript:605]: This should be renamed to `JustTookDamage` to emphasize at a glance that there's no hit target and that it's only ever true in the immediate past.

File: RA Scripts/Legend of Zelda, The The Minish Cap.rascript [hunk 3]
```diff
@@ -1205,4 +1207,3 @@ rich_presence_display("{0}{1}{2}{3}{4}{5} [❤️{6}] [🗽{7}] [🧿{8}]",
     rich_presence_value("Figurines", byte(0x00AAF0)),
     rich_presence_value("Fusions", byte(0x00AB57))
 )
-
```
Comment (MagmarFire) [inline RA Scripts/Legend of Zelda, The The Minish Cap.rascript:1208]: I recommend readding this line to maintain adherence to POSIX (and to reduce the total number of diffs in this PR).

**Review by MagmarFire — CHANGES_REQUESTED** [review]
My apologies for not seeing this PR until now!

File: RA Scripts/Legend of Zelda, The The Minish Cap.rascript (hunk 1 above)
Comment (MagmarFire) [reply RA Scripts/Legend of Zelda, The The Minish Cap.rascript:603]: Though I do acknowledge that the original was also in PascalCase. This was an error on my part.

**Comment by MagmarFire** [comment]
The new changes look good, but there are still unaddressed comments. I won't approve a merge until they're resolved.

**Comment by wolfman2000** [comment]
The unaddressed comment should be covered at this point unless I missed another location.

**Review by MagmarFire — APPROVED** [review]

**Pull request merged**